src/Adapter/Profile/Permission/QueryHandler/GetPermissionsForConfigurationHandler.php

```diff
@@ -0,0 +1,347 @@
+use PrestaShop\PrestaShop\Core\Domain\Profile\Permission\ValueObject\Permission;
+use PrestaShopBundle\Security\Voter\PageVoter;
+use Profile;
+use RuntimeException;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+use Tab;
+
+/**
+ * Get configurable permissions
+ *
+ * @internal
+ */
+final class GetPermissionsForConfigurationHandler implements GetPermissionsForConfigurationHandlerInterface
+{
+    /**
+     * @internal Max nesting level for building tabs tree
+     */
+    const MAX_NESTING_LEVEL = 12;
+
+    /**
+     * @var AuthorizationCheckerInterface
+     */
+    private $authorizationChecker;
+
+    /**
+     * @var array
+     */
+    private $whitelist = [];
+
+    /**
+     * @param AuthorizationCheckerInterface $authorizationChecker
+     */
+    public function __construct(AuthorizationCheckerInterface $authorizationChecker)
+    {
+        $this->authorizationChecker = $authorizationChecker;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(GetPermissionsForConfiguration $query): ConfigurablePermissions
+    {
+        $profiles = $this->getProfilesForPermissionsConfiguration();
+        $tabs = $this->getTabsForPermissionsConfiguration();
+        $permissions = [
+            Permission::VIEW,
+            Permission::ADD,
+            Permission::EDIT,
+            Permission::DELETE,
+        ];
+
+        $tabPermissionsForProfiles = $this->getTabPermissionsForProfiles($profiles);
+        $modulePermissionsForProfiles = $this->getModulePermissionsForProfiles($profiles);
+
+        $permissionIds = array_flip(Permission::SUPPORTED_PERMISSIONS);
+        $employeeProfileId = $query->getEmployeeProfileId()->getValue();
+        $canEmployeeEditPermissions = $this->authorizationChecker->isGranted(PageVoter::UPDATE, 'AdminAccess');
+
+        $bulkConfigurationPermissions = $this->getBulkConfigurationForProfiles(
+            $employeeProfileId,
+            $canEmployeeEditPermissions,
+            $tabPermissionsForProfiles,
+            $profiles,
+            $tabs,
+            $permissions
+        );
+
+        return new ConfigurablePermissions(
+            $tabPermissionsForProfiles,
+            $modulePermissionsForProfiles,
+            $profiles,
+            $tabs,
+            $bulkConfigurationPermissions,
+            $permissions,
+            $permissionIds,
+            $employeeProfileId,
+            $canEmployeeEditPermissions
+        );
+    }
+
+    /**
+     * @return int[] IDs of non configurable tabs
+     */
+    private function getNonConfigurableTabs(): array
+    {
+        return [
+            Tab::getIdFromClassName('AdminLogin'),
+            Tab::getIdFromClassName('AdminQuickAccesses'),
+        ];
+    }
+
+    /**
+     * @return array
+     */
+    private function getProfilesForPermissionsConfiguration(): array
+    {
+        $legacyProfiles = Profile::getProfiles(Context::getContext()->language->id);
```

Comment:
```suggestion
        $legacyProfiles = Profile::getProfiles(Context::getContext()->language->id);
```

Please push in the constructor : `Context::getContext()->language->id`